content/en/blog/_posts/2019-05-22-OPA-Gatekeeper-Policy-and-Governance-for-Kubernetes.md

```diff
@@ -0,0 +1,178 @@
+
+## Evolution 
+
+Before we dive into the current state of Gatekeeper, let’s take a look at how the Gatekeeper project has evolved. 
+
+* Gatekeeper v1.0 - Uses OPA as the admission controller with the kube-mgmt sidecar enforcing configmap-based policies. It provides validating and mutating admission control. Donated by Styra. 
+* Gatekeeper v2.0 - Uses Kubernetes policy controller as the admission controller with OPA and kube-mgmt sidecars enforcing configmap-based policies. It provides validating and mutating admission control and audit functionality. Donated by Microsoft. 
+* Gatekeeper v3.0 - The admission controller is integrated with the [OPA constraint framework](https://github.com/open-policy-agent/frameworks/tree/master/constraint) to enforce CRD-based policies and allow declaratively configured policies to be reliably shareable. Built with kubebuilder, it provides validating and mutating (to be implemented) admission control and audit functionality. This enables the creation of policy templates for [Rego](https://www.openpolicyagent.org/docs/v0.10.7/how-do-i-write-policies/) policies, creation of policies as CRDs, and storage of audit results on policy CRDs. This project is a collaboration between Google, Microsoft, Red Hat, and Styra. 
+
+![](/images/blog/2019-05-22-opa-gatekeeper/v3.png)
+ 
+## Gatekeeper v3.0 Features 
+
+Now let’s take a closer look at the current state of Gatekeeper and how you can leverage all the latest features. Consider an organization that wants to ensure all objects in a cluster have departmental information provided as part of the object’s labels. How do we do this with Gatekeeper? 
+
+### Validating Admission Control 
+
+Once all the Gatekeeper components have been [installed](https://github.com/open-policy-agent/gatekeeper) in your cluster, the API server will trigger the Gatekeeper admission webhook to process the admission request whenever a resource in the cluster is created, updated, or deleted. 
+
+During the validation process, Gatekeeper will enforce all policies executed by OPA. Each policy is written with [Rego](https://www.openpolicyagent.org/docs/v0.10.7/how-do-i-write-policies/), a declarative query language used by OPA to enumerate instances of data that violate the expected state of the system. All policies are evaluated as a logical AND. If one policy is not satisfied, then the whole request is rejected. 
```

Comment:
Technically the API Server does the enforcement. GK (g8r?) acts as a bridge between the API Server and OPA.

From the user's perspective, a good story might be to have them install a simple constraint, see it enforced, then create a policy that is a combination of constraints (introducing the AND-ing), finally gradually ease them in to writing their own ConstraintTemplates.

In any case, we should introduce the concept of constraints so a user has a basic understanding before we talk about how they interact (ANDing).

This may be solved by moving this section to after the next one (Policies and Constraints)?